EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
+                        ListItemStyle = ListDialogItem.ListItemStyle.Small;
+                        Buttons = ScrollingListDialogButtons.BackCancel;
+                        break;
+                    case GuildDialogState.WaitingForMembers:
+                        ListItemStyle = ListDialogItem.ListItemStyle.Small;
+                        Buttons = ScrollingListDialogButtons.BackCancel;
+                        break;
+                }
+            }
         }
 
         private readonly IEODialogIconService _dialogIconService;
         private readonly ILocalizedStringFinder _localizedStringFinder;
         private readonly ICharacterProvider _characterProvider;
         private readonly IEOMessageBoxFactory _messageBoxFactory;
+        private readonly ITextMultiInputDialogFactory _textMultiInputDialogFactory;
+        private readonly IContentProvider _contentProvider;
+        private readonly IGuildActions _guildActions;
+        private readonly IGuildSessionProvider _guildSessionProvider;
+        private readonly ICharacterInventoryProvider _characterInventoryProvider;
+        private readonly IEIFFileProvider _eifFileProvider;
 
-        private GuildDialogState _state;
+        private readonly Stack<State> _stateStack;
+        private int _creationMemberCount;
 
+        private State _state;
         public GuildDialog(INativeGraphicsManager nativeGraphicsManager,
                          IEODialogButtonService dialogButtonService,
                          IEODialogIconService dialogIconService,
                          ILocalizedStringFinder localizedStringFinder,
                          ICharacterProvider characterProvider,
-                         IEOMessageBoxFactory messageBoxFactory)
+                         IEOMessageBoxFactory messageBoxFactory,
+                         ITextMultiInputDialogFactory textMultiInputDialogFactory,
+                         IContentProvider contentProvider,
+                         IGuildActions guildActions,
+                         IGuildSessionProvider guildSessionProvider,
+                         ICharacterInventoryProvider characterInventoryProvider,
+                         IEIFFileProvider eifFileProvider)
             : base(nativeGraphicsManager, dialogButtonService, DialogType.Guild)
         {
             _dialogIconService = dialogIconService;
             _localizedStringFinder = localizedStringFinder;
             _characterProvider = characterProvider;
             _messageBoxFactory = messageBoxFactory;
+            _textMultiInputDialogFactory = textMultiInputDialogFactory;
+            _contentProvider = contentProvider;
+            _guildActions = guildActions;
+            _guildSessionProvider = guildSessionProvider;
+            _characterInventoryProvider = characterInventoryProvider;
+            _eifFileProvider = eifFileProvider;
+            _stateStack = new Stack<State>();
+            _creationMemberCount = 0;
```

Comment:
*Technically* this line of code is redundant since it'll be zero-initialized by default